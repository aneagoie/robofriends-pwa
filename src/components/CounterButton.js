import React, { Component } from "react";

class CounterButton extends Component {
  constructor(props) {
    super(props);
    this.state = {
      count: 0,
    };
  }

  render() {
    return (
      <button
        id="counter"
        color={this.props.color}
        onClick={() => this.setState({ count: this.state.count + 1 })}
      >
        Count: {this.state.count}
      </button>
    );
  }
}

export default CounterButton;
