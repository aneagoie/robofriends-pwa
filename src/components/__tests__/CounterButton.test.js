import { shallow } from "enzyme";
import React from "react";
import CounterButton from "../CounterButton";

test("Should render CounterButton component", () => {
  const mockColor = "red";
  expect(
    shallow(<CounterButton id={"counter"} color={mockColor} />)
  ).toMatchSnapshot();
});

test("correctly increments the counter", () => {
  const mockColor = "red";
  const mockButton = shallow(<CounterButton color={mockColor} />);

  mockButton.find('[id="counter"]').simulate("click");
  expect(mockButton.state()).toEqual({ count: 1 });
});
