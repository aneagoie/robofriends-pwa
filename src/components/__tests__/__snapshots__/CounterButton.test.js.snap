// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render CounterButton component 1`] = `
<button
  color="red"
  id="counter"
  onClick={[Function]}
>
  Count: 
  0
</button>
`;
